examples/stable_diffusion_xl/gm/data/dataset_wds.py

```diff
@@ -245,9 +246,12 @@ def __init__(self, shardlist_desc=None, *args, **kwargs):
 
         with open(shardlist_desc, "r") as fp:
             shardlist = json.load(fp)["shardlist"]
-        self.dataset = wids.ShardListDataset(shardlist)
+        self.dataset = wids.ShardListDataset(shardlist, cache_dir=cache_dir)
         self._datalen = len(self.dataset)
 
+    def __exit__(self, exc_type, exc_val, exc_tb):
```

Comment:
`__exit__` is used for context managers (i.e. `with` statement). It should be `__del__` here.